packages/jest-matchers/src/matchers.js

```diff
@@ -573,11 +573,13 @@ const matchers: MatchersObject = {
           return false;
         }
 
-        return expected.every(exp => {
-          return received.some(act => {
-            return compare(exp, act);
-          });
-        });
+        for (let i = 0; i < expected.length; i++) {
```

Comment:
If you are interested, an array method is alternative to `for` loop here in `compare`

```js
return expected.every((exp, i) => compare(exp, received[i]));
```

and also in corresponding logic of `findMatchObject`

```js
return expected.map((exp, i) => findMatchObject(exp, received[i]));
```